float32[] sensor_data
---
string action